packages/components/src/border-box-control/border-box-control-split-controls/component.tsx

```diff
@@ -36,16 +36,25 @@ const BorderBoxControlSplitControls = (
 		__next36pxDefaultSize,
 		...otherProps
 	} = useBorderBoxControlSplitControls( props );
-	const containerRef = useRef();
-	const mergedRef = useMergeRefs( [ containerRef, forwardedRef ] );
-	const popoverProps = popoverPlacement
-		? {
+	const [ popoverProps, setPopoverProps ] = useState< PopoverPartialProps >();
+	const [ popoverAnchor, setPopoverAnchor ] = useState< Element >();
+
+	const containerRef = useCallback( ( node ) => {
+		setPopoverAnchor( node );
+	}, [] );
+
+	useEffect( () => {
+		if ( popoverPlacement ) {
+			setPopoverProps( {
 				placement: popoverPlacement,
 				offset: popoverOffset,
-				anchorRef: containerRef,
+				anchor: popoverAnchor,
 				__unstableShift: true,
-		  }
-		: undefined;
+			} );
+		} else {
+			setPopoverProps( undefined );
+		}
+	}, [ popoverPlacement, popoverOffset, popoverAnchor ] );
```

Comment:
This changes are in place to make sure that, when the `containerRef` is updated, `popoverProps` are updated too.

Without it, the ref would be updated _after_ the initial value of `popoverProps` was computed, and because of how React refs work, React wouldn't re-render the component — meaning that the `anchor` property in `popoverProps` would be stale until a re-render would be caused by whatever reason.